File: 1807-PartitioningIntoMinimumNumberOfDeciBinaryNumbers/1807-PartitioningIntoMinimumNumberOfDeciBinaryNumbers.java
// Last updated: 3/2/2026, 1:58:58 PM
class Solution {
    public int minPartitions(String n) {
        int maxdigit = -1;
        for(char ch : n.toCharArray()){
            int digit = ch - '0';
            if(digit>maxdigit) maxdigit = digit;
        }
        return maxdigit;
    }
}

File: 1807-PartitioningIntoMinimumNumberOfDeciBinaryNumbers/1807-PartitioningIntoMinimumNumberOfDeciBinaryNumbers.cpp
// Last updated: 3/13/2026, 8:16:38 PM
class Solution {
public:
    int minPartitions(string n) {
        int maxdigit = 0;
        for(char  ch : n){
            maxdigit = max(maxdigit,ch-'0');
        }
        return maxdigit;
    }
};